### PR TITLE
2025-insecure-authentication

### PR DESCRIPTION
**Pull request by Tomer-Yahalom**
This commit adds the NHI4:2025 Insecure Authentication risk page.
It includes the ranking criteria table, description, example scenarios, preventative measures, references and data points used for ranking.

### CODE REVIEW

File: 2025/docs/4-insecure-authentication.md [hunk 1]
```diff
@@ -1 +1,44 @@
-TBA
\ No newline at end of file
+# NHI4:2025 Insecure Authentication
+
+| Threat Agents & Attack Vectors                    | Security Weakness                                                                                          | Impact                                         |
+|---------------------------------------------------|-------------------------------------------------------------------------------------------------------------|------------------------------------------------|
+| Exploitability: **Easy**            | Prevalence: **Widespread**<br>Detectability: **Hard**                       | Technical Impact: **Severe**     |
```
Comment (torchsec-roni) [inline 2025/docs/4-insecure-authentication.md:5]: Detectability is marked as "Easy" in the doc: https://docs.google.com/document/d/18Bu2ixzbxWFP-OBt7ZeOojBsOQ9daqUc8jw6uETbOO0/edit?tab=t.0#heading=h.mrv1n2cxy4m7

Comment (torchsec-roni) [reply 2025/docs/4-insecure-authentication.md:5]: Also, Technical Impact is marked as "Moderate" in the doc: https://docs.google.com/document/d/18Bu2ixzbxWFP-OBt7ZeOojBsOQ9daqUc8jw6uETbOO0/edit?tab=t.0#heading=h.mrv1n2cxy4m7

Comment (torchsec-roni) [reply 2025/docs/4-insecure-authentication.md:5]: Please also add "Business Impact: Specific" under "Impact"

Comment (Tomer-Yahalom) [reply 2025/docs/4-insecure-authentication.md]: Don't know how i confused those, Thanks!
Opened another commit with the fixes 👍

**Review by ronilichtman — APPROVED** [review]
LGTM! 🔥

**Pull request merged**